### PR TITLE
Added document for each function and for each properties.

### LINKED ISSUE #7: documents-for-functions-and-properties
- Add document with 'option + command + slash' to essential and complex properties and functions
- No need for insignificant function and properties

### PR DESCRIPTION
**Pull request by ferhanakkan**
For all properties and functions added document descriptions.

### CODE REVIEW

File: Example/Tom.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>IDEDidComputeMac32BitWarning</key>
+	<true/>
+</dict>
+</plist>
```
Comment (meyusufdemirci) [inline Example/Tom.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Why did you add this file? If you are not interested in it, please remove it.

File: Tom/Classes/Configuration.swift [hunk 2]
```diff
@@ -5,13 +5,20 @@
...
 
+/// Configuration parameters for TomView.
 public struct Configuration {
-
+    
+    /// Determines how many lines will be in TomView.
     public let lineCount: Int
+    /// Determines the color of the lines in TomView.
     public let lineColor: UIColor
+    /// Determines spacing behind each line.
```
Comment (meyusufdemirci) [inline Tom/Classes/Configuration.swift]:
```suggestion
    /// Determines spacing between each line.
```

File: Tom/Classes/Configuration.swift [hunk 3]
```diff
@@ -5,13 +5,20 @@
...
     public let lineSpacing: CGFloat
+    /// Determines axis of lines.
```
Comment (meyusufdemirci) [inline Tom/Classes/Configuration.swift]: Could you please add the values either? Something like that,
```suggestion
    /// Determines axis of lines, horizontal or vertical.
```

File: Tom/Classes/Configuration.swift [hunk 4]
```diff
@@ -5,13 +5,20 @@
...
     public let axis: UILayoutConstraintAxis = .horizontal
+    /// Determines minimum line height in TomView
     public let minLineHeight: CGFloat
+    /// Determines animation duraction for line animations.
```
Comment (meyusufdemirci) [inline Tom/Classes/Configuration.swift]:
```suggestion
    /// Determines animation duration for the line animation.
```

File: Tom/Classes/CornerRadius.swift [hunk 5]
```diff
@@ -7,7 +7,10 @@
 
 import Foundation
 
+/// Keys that specify a corner radius type.
 enum CornerRadius {
+    /// Keys that specify corner radius half of view's height.
     case rounded
+    /// Keys that specify corner radius for developer selected radius
     case custom(CGFloat)
 }
```
Comment (meyusufdemirci) [inline Tom/Classes/CornerRadius.swift]: I think no need any comment for this class because nobody will able to see it from the property summary

File: Tom/Classes/LineConfiguration.swift [hunk 6]
```diff
@@ -5,8 +5,11 @@
 //  Created by Yusuf Demirci on 30.03.21.
 //
 
+/// Line configuration parameters.
```
Comment (meyusufdemirci) [inline Tom/Classes/LineConfiguration.swift]: I think no need it since the struct name is pretty clear

File: Tom/Classes/TomView.swift [hunk 7]
```diff
@@ -15,9 +15,12 @@ final public class TomView: UIView {
         return view
     }()
 
+    /// Array that contains the lines within itself.
     private var lineViews: [LineView] = []
+    /// Represent the state of the animation.
```
Comment (meyusufdemirci) [inline Tom/Classes/TomView.swift]: No need comment for private properties

File: Tom/Classes/TomView.swift [hunk 8]
```diff
@@ -15,9 +15,12 @@ final public class TomView: UIView {
         return view
     }()
 
+    /// Array that contains the lines within itself.
```
Comment (meyusufdemirci) [inline Tom/Classes/TomView.swift]: No need for privates

File: Tom/Classes/TomView.swift [hunk 9]
```diff
@@ -15,9 +15,12 @@ final public class TomView: UIView {
...
     private var lineViews: [LineView] = []
+    /// Represent the state of the animation.
     private var isAnimationEnabled: Bool = false
 
+    /// Configuration parameters for TomView.
```
Comment (meyusufdemirci) [inline Tom/Classes/TomView.swift]: No need

File: Tom/Classes/TomView.swift [hunk 10]
```diff
@@ -62,6 +67,7 @@ public extension TomView {
 
 private extension TomView {
 
+    /// Set configuration for TomView.
```
Comment (meyusufdemirci) [inline Tom/Classes/TomView.swift]: No need for privates

File: Tom/Classes/TomView.swift [hunk 11]
```diff
@@ -72,6 +78,8 @@ private extension TomView {
         addLinesToStackView()
     }
 
+    /// Start infitine animation while isAnimatonEnable parameter is true.
+    /// - Parameter lineView: Input parameter is type of UIView object or subobject of UIView type.
```
Comment (meyusufdemirci) [inline Tom/Classes/TomView.swift]: No need for privates

File: Tom/Classes/TomView.swift [hunk 12]
```diff
@@ -87,6 +95,7 @@ private extension TomView {
         }
     }
 
+    /// Creates a LineView as many line count in the configuration.
```
Comment (meyusufdemirci) [inline Tom/Classes/TomView.swift]: No need for privates

File: Tom/Classes/TomView.swift [hunk 13]
```diff
@@ -106,6 +115,9 @@ private extension TomView {
         }
     }
 
+    /// Pins lineView edges to containerView.
+    /// - Parameter lineView: Input parameter is type of UIView object or subobject of UIView type.
+    /// - Parameter containerView: Input parameter is type of UIView object or subobject of UIView type.
```
Comment (meyusufdemirci) [inline Tom/Classes/TomView.swift]: No need for privates

File: Tom/Classes/UIView+Extension.swift [hunk 14]
```diff
@@ -6,14 +6,17 @@
...
+    /// Returns height contraint of related view.
     var heightConstraint: NSLayoutConstraint? {
         constraints.first { $0.firstAttribute == .height && $0.relation == .equal }
     }
 }
 
 extension UIView {
 
+    /// Pins edges to methods input view.
+    /// - Parameter view: Input parameter is type of UIView object or subobject of UIView type.
```
Comment (meyusufdemirci) [inline Tom/Classes/UIView+Extension.swift]: Since it uses in only the pod, no need a comment

File: Tom/Classes/UIView+Extension.swift [hunk 15]
```diff
@@ -6,14 +6,17 @@
 //
 
 extension UIView {
-
+    
+    /// Returns height contraint of related view.
```
Comment (meyusufdemirci) [inline Tom/Classes/UIView+Extension.swift]: Since it uses in only the pod, no need a comment

File: Example/Tom.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (hunk 1 above)
Comment (ferhanakkan) [reply Example/Tom.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Actually Xcode added it automatically. When I looked documents its look like better to commit this file. Because it will safe a time while building example project.

Source => https://stackoverflow.com/questions/50367896/ideworkspacechecks-plist-file-suddenly-appear-after-updated-xcode

Comment (ferhanakkan) [reply Example/Tom.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Removed

**Review by meyusufdemirci — APPROVED** [review]
